fabric-keybindings-v0/src/main/java/net/fabricmc/fabric/api/client/keybinding/FabricKeyBinding.java

```diff
@@ -41,18 +61,66 @@ protected FabricKeyBinding(Identifier id, InputUtil.Type type, int code, String
 	}
 
 	public static class Builder {
-		protected final FabricKeyBinding binding;
 
-		protected Builder(FabricKeyBinding binding) {
-			this.binding = binding;
+		private InputUtil.Type type = InputUtil.Type.KEYSYM;
+
+		private Identifier id = null;
+
+		private int code = GLFW.GLFW_KEY_UNKNOWN;
+
+		private String category = KeyCategory.MISC;
+
+		private Builder() { }
+
+		public Builder id(Identifier keyName) {
+			this.id = keyName;
+			return this;
+		}
+
+		public Builder category(String category) {
+			this.category = category;
+			return this;
+		}
+
+		public Builder code(int keyCode) {
+			this.code = keyCode;
+			return this;
+		}
+
+		public Builder type(InputUtil.Type type) {
+			this.type = type;
+			return this;
 		}
 
 		public FabricKeyBinding build() {
-			return binding;
+			Objects.requireNonNull(id, "Keybindings should be created with an identifier.");
+
+			return new FabricKeyBinding(id, type, code, category);
+		}
+
+		public static Builder create() {
```

Comment:
can you move this to FabricKeybinding.builder instead of FabricKeybindingBuilder.create?

Reply:
Isn't it the norm to do `FabricKeybinding.Builder.create()....build()`?

Unfortunately, I didn't have the luxury to name the original builder class...

Reply:
I most often see `FabricKeyBinding.builder()...build()`. That's the pattern adopted by guava for example, as well as the dimensions API.

Reply:
Alrighty. I'll switch to that.